app/utils/callback/lava.py

```diff
@@ -85,291 +69,249 @@
-        meta[models.BOOT_RESULT_KEY] = "UNKNOWN"
-
-
-def _get_lava_boot_meta(meta, boot_meta):
-    """Parse the boot and login meta-data from LAVA
-
-    :param meta: The boot meta-data.
-    :type meta: dictionary
-    :param boot_meta: The boot and auto_login meta-data from the LAVA v2 job.
-    :type boot_meta: dictionary
-    """
-    meta[models.BOOT_TIME_KEY] = boot_meta["duration"]
-    extra = boot_meta.get("extra", None)
-    if extra is None:
-        return
-    kernel_messages = []
-    for e in extra:
-        fail = e.get("fail", None)
-        if not fail:
-            continue
-        if isinstance(fail, str):
-            kernel_messages.append(fail)
-        else:
-            for msg in (f.get("message", None) for f in fail):
-                if msg:
-                    kernel_messages.append(msg)
-    if kernel_messages:
-        meta[models.BOOT_WARNINGS_KEY] = len(kernel_messages)
-
-
-def _get_lava_bootloader_meta(meta, bl_meta):
-    """Parse the bootloader meta-data from LAVA
-
-    :param meta: The boot meta-data.
-    :type meta: dictionary
-    :param bl_meta: The bootloader meta-data from the LAVA v2 job.
-    :type bl_meta: dictionary
-    """
-    extra = bl_meta.get("extra", None)
-    if extra is None:
-        return
-    for e in extra:
-        for k, v in e.iteritems():
-            meta_key = BL_META_MAP.get(k, None)
-            if meta_key:
-                meta[meta_key] = v
-
-
-def _get_lava_meta(meta, job_data):
-    """Parse the meta-data from LAVA
-
-    Go through the LAVA meta-data and extract the fields needed to create a
-    boot entry in the database.
-
-    :param meta: The boot meta-data.
-    :type meta: dictionary
-    :param job_data: The JSON data from the callback.
-    :type job_data: dict
-    """
-    lava = yaml.load(job_data["results"]["lava"], Loader=yaml.CLoader)
-    meta_handlers = {
-        'job': _get_lava_job_meta,
-        'auto-login-action': _get_lava_boot_meta,
-        'bootloader-overlay': _get_lava_bootloader_meta,
+class LavaCallback(object):
+    META_DATA_MAP_TEST = {
+        models.ARCHITECTURE_KEY: "job.arch",
+        models.DEFCONFIG_KEY: "kernel.defconfig",
+        models.DEFCONFIG_FULL_KEY: "kernel.defconfig_full",
+        models.DEVICE_TYPE_KEY: "device.type",
+        models.DTB_KEY: "platform.dtb",
+        models.ENDIANNESS_KEY: "kernel.endian",
+        models.GIT_BRANCH_KEY: "git.branch",
+        models.GIT_COMMIT_KEY: "git.commit",
+        models.GIT_DESCRIBE_KEY: "git.describe",
+        models.GIT_URL_KEY: "git.url",
+        models.INITRD_KEY: "job.initrd_url",
+        models.JOB_KEY: "kernel.tree",
+        models.KERNEL_KEY: "kernel.version",
+        models.KERNEL_IMAGE_KEY: "job.kernel_image",
+        models.MACH_KEY: "platform.mach",
+        models.PLAN_KEY: "test.plan",
+        models.PLAN_VARIANT_KEY: "test.plan_variant",
+        models.BUILD_ENVIRONMENT_KEY: "job.build_environment",
```

Comment:
The list is almost in alphabetical order, maybe move these 2 to make it completely?

Reply:
Corrected